feather/server/src/packet_handlers/entity_action.rs

```diff
@@ -36,10 +39,20 @@ pub fn handle_entity_action(game: &mut Game, player: Entity, packet: EntityActio
             // a notice that bed state might have changed.
         }
         EntityActionKind::StartSprinting => {
-            //TODO issue #423
+            let is_sprinting = game.ecs.get_mut::<Sprinting>(player)?.0;
```

Comment:
This can be made into a single match branch to reduce code duplication.
```rust
EntityActionKind::StartSprinting |  EntityActionKind::StopSprinting => {
  let start_sprinting = matches!(EntityActionKind::StartSprinting, packet.action_id);
  ...
}
````

Reply:
Updated, based on your explaination

Reply:
Hold on, the analyzer doesn't like something

Reply:
What's the error? Maybe try `packet.action_id == EntityActionKind::StartSprinting`?

Reply:
(Meant to reply here but couldn't for some reason so I replied further down)